pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -858,105 +858,163 @@ func TestPriorityQueue_Pop(t *testing.T) {
 }
 
 func TestPriorityQueue_Update(t *testing.T) {
-	objs := []runtime.Object{highPriorityPodInfo.Pod, unschedulablePodInfo.Pod, medPriorityPodInfo.Pod}
 	c := testingclock.NewFakeClock(time.Now())
 	logger, ctx := ktesting.NewTestContext(t)
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
-	q := NewTestQueueWithObjects(ctx, newDefaultQueueSort(), objs, WithClock(c))
-	// add highPriorityPodInfo to activeQ.
-	q.Update(logger, nil, highPriorityPodInfo.Pod)
-	if _, exists, _ := q.activeQ.Get(newQueuedPodInfoForLookup(highPriorityPodInfo.Pod)); !exists {
-		t.Errorf("Expected %v to be added to activeQ.", highPriorityPodInfo.Pod.Name)
-	}
-	if len(q.nominator.nominatedPods) != 0 {
-		t.Errorf("Expected nomindatePods to be empty: %v", q.nominator)
-	}
-	// Update highPriorityPodInfo and add a nominatedNodeName to it.
-	q.Update(logger, highPriorityPodInfo.Pod, highPriNominatedPodInfo.Pod)
-	if q.activeQ.Len() != 1 {
-		t.Error("Expected only one item in activeQ.")
-	}
-	if len(q.nominator.nominatedPods) != 1 {
-		t.Errorf("Expected one item in nomindatePods map: %v", q.nominator)
-	}
-	// Updating an unschedulable pod which is not in any of the two queues, should
-	// add the pod to activeQ.
-	q.Update(logger, unschedulablePodInfo.Pod, unschedulablePodInfo.Pod)
-	if _, exists, _ := q.activeQ.Get(newQueuedPodInfoForLookup(unschedulablePodInfo.Pod)); !exists {
-		t.Errorf("Expected %v to be added to activeQ.", unschedulablePodInfo.Pod.Name)
-	}
-	// Updating a pod that is already in activeQ, should not change it.
-	q.Update(logger, unschedulablePodInfo.Pod, unschedulablePodInfo.Pod)
-	if len(q.unschedulablePods.podInfoMap) != 0 {
-		t.Error("Expected unschedulablePods to be empty.")
-	}
-	if _, exists, _ := q.activeQ.Get(newQueuedPodInfoForLookup(unschedulablePodInfo.Pod)); !exists {
-		t.Errorf("Expected: %v to be added to activeQ.", unschedulablePodInfo.Pod.Name)
-	}
-	if p, err := q.Pop(); err != nil || p.Pod != highPriNominatedPodInfo.Pod {
-		t.Errorf("Expected: %v after Pop, but got: %v", highPriorityPodInfo.Pod.Name, p.Pod.Name)
-	}
 
-	// Updating a pod that is in backoff queue and it is still backing off
-	// pod will not be moved to active queue, and it will be updated in backoff queue
-	podInfo := q.newQueuedPodInfo(medPriorityPodInfo.Pod)
-	if err := q.podBackoffQ.Add(podInfo); err != nil {
-		t.Errorf("adding pod to backoff queue error: %v", err)
-	}
-	q.Update(logger, podInfo.Pod, podInfo.Pod)
-	rawPodInfo, err := q.podBackoffQ.Pop()
-	podGotFromBackoffQ := rawPodInfo.(*framework.QueuedPodInfo).Pod
-	if err != nil || podGotFromBackoffQ != medPriorityPodInfo.Pod {
-		t.Errorf("Expected: %v after Pop, but got: %v", medPriorityPodInfo.Pod.Name, podGotFromBackoffQ.Name)
-	}
+	tests := []struct {
+		name         string
+		prepareFunc  func(q *PriorityQueue) (oldPod, newPod *v1.Pod)
+		validateFunc func(q *PriorityQueue, expectedPod *v1.Pod)
```

Comment:
Now, most of validateFunc is just checking where the target Pod was moved + checking nominator. 

Can we check where the target Pod was moved by like adding `expectedQ` like this test:
https://github.com/kubernetes/kubernetes/blob/70aa0f87499179e985e73c489444c92aa4a5dc01/pkg/scheduler/internal/queue/scheduling_queue_test.go#L1399-L1409

Also, I believe we can remove nominator checking from `validateFunc` somehow as well.
It'd be nice if you could remove all of `validateFunc` eventually.